src/server/models/listItem.ts

```diff
@@ -195,6 +204,143 @@ export async function checkListItemExists({
   return total > 0;
 }
 
+export async function getPaginationValues(  props: {
+  count: number;
+  page: number;
+  params: ListsRequestParams;
+}): Promise<PaginationResults> {
+  const { count, page, params } = props;
+  const limit = 20;
+  let allPages = 0;
+  if (count > 0 && limit > 0) {
+    allPages = Math.ceil(count / limit);
+  }
+
+  const nextPrevious = await getNextPrevious({page, allPages, params});
+  const { queryString, currentPage } = nextPrevious;
+
+  const pageItems: PaginationItem[] = await getPageItems({allPages, currentPage, queryString});
+
+  const from = await getFromCount({count, limit, currentPage});
+  const to = await getToCount({currentPage, allPages, count, limit});
+
+  return {
+    pagination: {
+      results: {
+        from,
+        to,
+        count,
+        currentPage
+      },
+      previous: {
+        text: nextPrevious.previous.page.toString(),
+        href: nextPrevious.previous.queryString
+      },
+      next: {
+        text: nextPrevious.next.page.toString(),
+        href: nextPrevious.next.queryString
+      },
+      items: pageItems
+    }
+  }
+}
+
+ async function getNextPrevious(props: {page: number; allPages: number; params: ListsRequestParams }): Promise<{
```

Comment:
spent too much time in react? 😄 This is recommended for react but otherwise it's pretty common to destructure the props here. I would say that they are params rather than props in this context as well.  

```suggestion
type getPaginationParams = { count: number; page: number; params: number;}

 async function getNextPrevious({count, page, params}: getPaginationParams ): Promise<..>  {
// count, page, params should be available to you now, no need to do const { ... } = props
 }
```

Reply:
if you can call the ETD project react!  Refactored to this pattern.